mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -268,6 +294,10 @@ class ALSModel private[ml] (
 @Since("1.6.0")
 object ALSModel extends MLReadable[ALSModel] {
 
+  private val NaN = "nan"
+  private val Drop = "drop"
+  private[recommendation] final val supportedUnknownStrategies = Array(NaN, Drop)
```

Comment:
Instead of adding an array here, could you just create it in the `ParamValidator` when defining the `Param`?
```scala
ParamValidators.inArray(Array(ALSModel.NaN, ALSModel.Drop))
```